src/main/java/carpet/mixins/BarrierBlock_updateSuppressionBlockMixin.java

```diff
@@ -0,0 +1,55 @@
+package carpet.mixins;
+
+import carpet.CarpetSettings;
+import net.minecraft.block.BarrierBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.PoweredRailBlock;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+
+import java.util.Random;
+
+@Mixin(BarrierBlock.class)
+public class BarrierBlock_updateSuppressionBlockMixin extends Block {
+    private boolean shouldPower = false;
+
+    public BarrierBlock_updateSuppressionBlockMixin(Settings settings) { super(settings); }
+
+    @Override
+    public int getWeakRedstonePower(BlockState state, BlockView world, BlockPos pos, Direction direction) {
```

Comment:
Why you are changing this behaviour unconditinally?

Reply:
getWeakRedstonePower does not exist in the BarrierBlock so I need to override it. shouldPower can only be changed within functions that have rule checks. Therefore it should keep its normal behavior. Unless you meant something else?